code/modules/shuttle/shuttle.dm

```diff
@@ -552,6 +552,16 @@
 	else
 		. = null
 
+/obj/docking_port/mobile/proc/check_exile_pass()
+	if (is_station_level(z))
+		return TRUE
+	for(var/place in shuttle_areas)
+		var/area/shuttle/shuttle_area = place
+		for(var/mob/living/L in shuttle_area)
```

Comment:
Does hiding in a crate make you leave the area loc?

Reply:
Yea. I did this because I thought it would be cheaper than checking every single mob, but it doesn't get the desired results.

I went ahead and added 2 solutions. Which one do you think will work best?

Reply:
Go through every single mob, check if they have an exile implant and then check if they are in the shuttle areas.

Reply:
Wait; is the exile implant check more expensive then the area check?